python/amici/petab_objective.py

```diff
@@ -628,6 +740,106 @@ def _get_measurements_and_sigmas(
+
+    if not measurement_ids:
+        # no steady state conditions found. Return an empty dataframe.
+        return pd.DataFrame(columns=('preequilibrationConditionId',
+                                     'simulationConditionId'),
+                            index=[], data=[])
+
+    # get simulation conditions with timepoint 'inf' make unique
+    steadystate_conditions = petab_problem.measurement_df.loc[measurement_ids,
+        ('preequilibrationConditionId', 'simulationConditionId')]
+    steadystate_conditions = steadystate_conditions.drop_duplicates()
+
+    # pandas sometimes reads in "NaN" if a cell is empty. Correct that.
+    for i_cond in steadystate_conditions.index:
+        tmp = steadystate_conditions.loc[i_cond, 'preequilibrationConditionId']
+        if type(tmp) == float and np.isnan(tmp):
+                steadystate_conditions.loc[i_cond,
+                                           'preequilibrationConditionId'] = ''
+
+    steadystate_conditions.drop_duplicates(inplace=True)
+
+    return steadystate_conditions
+
+
+def _filter_simulation_conditions(simulation_conditions: pd.DataFrame,
+                                  steadystate_conditions: pd.DataFrame,
+    ) -> pd.DataFrame:
+    """
+    Filter simulation conditions passed by user for steadystate data
+
+    :param simulation_conditions:
+        dataframe with simulation conditions passed by user
+
+    :param steadystate_conditions:
+        ddataframe with simulation conditions with all steadystate measurements
+
+    :return:
+        dataframe with conditions passed by user restricted to having
+        steadystate measurements
+    """
+    steadystate_tuples = [
+        (steadystate_conditions.loc[id, 'preequilibrationConditionId'],
+         steadystate_conditions.loc[id, 'simulationConditionId'])
+        for id in list(steadystate_conditions.index)
+    ]
+
+    # to avoid if-else-switches due to (non-)existing preequilibrationCondition
+    # column: Add a column, if not yet there
+    if 'preequilibrationConditionId' not in simulation_conditions.columns:
+        simulation_conditions['preequilibrationConditionId'] = \
+            [''] * simulation_conditions.shape[0]
+
+    # if conditions were passed drop those without steadystate measurements
+    for id in list(simulation_conditions.index):
+        if (simulation_conditions.loc[id, 'preequilibrationConditionId'],
+            simulation_conditions.loc[id, 'simulationConditionId']) not in \
+                steadystate_tuples:
+            simulation_conditions.drop(id, axis=0, inplace=True)
+
+    return simulation_conditions
+
+
+def _create_condition_list(simulation_conditions: pd.DataFrame) -> List[str]:
+    """
+    Create a list identifying the preequilibration and simulation conditions
+
+    :param simulation_conditions:
+        dataframe with simulation conditions passed by user
+
+    :return:
+        list with the preequilibration and simulation conditions
+    """
```

Comment:
Hm. Without reading the code, it's not really clear to me what goes in and what comes out. Required / provided columns, etc.